## Importamos las librerias

In [1]:
import pandas as pd
from datetime import date
import numpy as np

## Cargamos los CSV

In [2]:
cv1 = pd.read_csv(filepath_or_buffer="Datos_usuario/1.csv",header=0, usecols=["userId","rating","timestamp","movieId"])
cv2 = pd.read_csv(filepath_or_buffer="Datos_usuario/2.csv",header=0, usecols=["userId","rating","timestamp","movieId"])
cv3 = pd.read_csv(filepath_or_buffer="Datos_usuario/3.csv",header=0, usecols=["userId","rating","timestamp","movieId"])
cv4 = pd.read_csv(filepath_or_buffer="Datos_usuario/4.csv",header=0, usecols=["userId","rating","timestamp","movieId"])
cv5 = pd.read_csv(filepath_or_buffer="Datos_usuario/5.csv",header=0, usecols=["userId","rating","timestamp","movieId"])
cv6 = pd.read_csv(filepath_or_buffer="Datos_usuario/6.csv",header=0, usecols=["userId","rating","timestamp","movieId"])
cv7 = pd.read_csv(filepath_or_buffer="Datos_usuario/7.csv",header=0, usecols=["userId","rating","timestamp","movieId"])
cv8 = pd.read_csv(filepath_or_buffer="Datos_usuario/8.csv",header=0, usecols=["userId","rating","timestamp","movieId"])

## Visualizamos la info de los DataFrames

In [3]:
cv1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500000 entries, 0 to 1499999
Data columns (total 4 columns):
 #   Column     Non-Null Count    Dtype  
---  ------     --------------    -----  
 0   userId     1500000 non-null  int64  
 1   rating     1500000 non-null  float64
 2   timestamp  1500000 non-null  int64  
 3   movieId    1500000 non-null  object 
dtypes: float64(1), int64(2), object(1)
memory usage: 45.8+ MB


In [4]:
cv2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500000 entries, 0 to 1499999
Data columns (total 4 columns):
 #   Column     Non-Null Count    Dtype  
---  ------     --------------    -----  
 0   userId     1500000 non-null  int64  
 1   rating     1500000 non-null  float64
 2   timestamp  1500000 non-null  int64  
 3   movieId    1500000 non-null  object 
dtypes: float64(1), int64(2), object(1)
memory usage: 45.8+ MB


In [5]:
cv3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500000 entries, 0 to 1499999
Data columns (total 4 columns):
 #   Column     Non-Null Count    Dtype  
---  ------     --------------    -----  
 0   userId     1500000 non-null  int64  
 1   rating     1500000 non-null  float64
 2   timestamp  1500000 non-null  int64  
 3   movieId    1500000 non-null  object 
dtypes: float64(1), int64(2), object(1)
memory usage: 45.8+ MB


In [6]:
cv4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500000 entries, 0 to 1499999
Data columns (total 4 columns):
 #   Column     Non-Null Count    Dtype  
---  ------     --------------    -----  
 0   userId     1500000 non-null  int64  
 1   rating     1500000 non-null  float64
 2   timestamp  1500000 non-null  int64  
 3   movieId    1500000 non-null  object 
dtypes: float64(1), int64(2), object(1)
memory usage: 45.8+ MB


In [7]:
cv5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500000 entries, 0 to 1499999
Data columns (total 4 columns):
 #   Column     Non-Null Count    Dtype  
---  ------     --------------    -----  
 0   userId     1500000 non-null  int64  
 1   rating     1500000 non-null  float64
 2   timestamp  1500000 non-null  int64  
 3   movieId    1500000 non-null  object 
dtypes: float64(1), int64(2), object(1)
memory usage: 45.8+ MB


In [8]:
cv6.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500000 entries, 0 to 1499999
Data columns (total 4 columns):
 #   Column     Non-Null Count    Dtype  
---  ------     --------------    -----  
 0   userId     1500000 non-null  int64  
 1   rating     1500000 non-null  float64
 2   timestamp  1500000 non-null  int64  
 3   movieId    1500000 non-null  object 
dtypes: float64(1), int64(2), object(1)
memory usage: 45.8+ MB


In [9]:
cv7.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 524289 entries, 0 to 524288
Data columns (total 4 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   userId     524289 non-null  int64  
 1   rating     524289 non-null  float64
 2   timestamp  524289 non-null  int64  
 3   movieId    524289 non-null  object 
dtypes: float64(1), int64(2), object(1)
memory usage: 16.0+ MB


In [10]:
cv8.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500000 entries, 0 to 1499999
Data columns (total 4 columns):
 #   Column     Non-Null Count    Dtype  
---  ------     --------------    -----  
 0   userId     1500000 non-null  int64  
 1   rating     1500000 non-null  float64
 2   timestamp  1500000 non-null  int64  
 3   movieId    1500000 non-null  object 
dtypes: float64(1), int64(2), object(1)
memory usage: 45.8+ MB


## Unimos todos los DataFrames

In [11]:
all_csv = pd.concat([cv1,cv2,cv3,cv4,cv5,cv6,cv7,cv8], axis = 0, join = "outer", ignore_index=True)

In [12]:
all_csv.shape # Vemos la cantidad de filas y de columnas

(11024289, 4)

## Cambiamos el nombre del campo "rating" por "score"

In [13]:
all_csv.rename(columns={"rating":"score"}, inplace=True)

### Visualizamos como son los datos

In [14]:
all_csv.head(5)

userId  score   timestamp movieId
0       1    1.0  1425941529   as680
1       1    4.5  1425942435  ns2186
2       1    5.0  1425941523  hs2381
3       1    5.0  1425941546  ns3663
4       1    5.0  1425941556  as9500

In [15]:
all_csv.tail(5)

userId  score   timestamp movieId
11024284  124380    4.5  1196786159  ns5272
11024285  124380    2.5  1196786030  ns5492
11024286  124380    3.5  1196785679   hs305
11024287  124380    4.5  1196787089  ns7881
11024288  124380    1.5  1196785847   as883

### Genero un DataFrame con datos agrupados por userId para conocer un poco sobre la cantidad de resenas que hicieron 

In [16]:
cantidad = all_csv.groupby("userId").count()
cantidad.drop(["timestamp", "movieId"], axis=1, inplace=True)
cantidad.rename(columns={"score":"cantidad_resenas"}, inplace=True)
cantidad.head()

cantidad_resenas
userId                  
1                     27
2                     22
3                     10
4                     62
5                     26

In [17]:
cantidad.max()

cantidad_resenas    18276
dtype: int64

In [18]:
cantidad[cantidad["cantidad_resenas"] == 18276]

cantidad_resenas
userId                  
45811              18276

In [19]:
cantidad[cantidad["cantidad_resenas"] == 1].count()  # cantidad de usuarios que solo han hecho una resena

cantidad_resenas    2129
dtype: int64

In [20]:
cantidad[cantidad["cantidad_resenas"] == 2].count() # cantidad de usuarios que solo han hecho dos resena

cantidad_resenas    1529
dtype: int64

In [21]:
cantidad[cantidad["cantidad_resenas"] == 3].count() # cantidad de usuarios que solo han hecho tres resena

cantidad_resenas    1439
dtype: int64

In [22]:
cantidad[cantidad["cantidad_resenas"] == 4].count() # cantidad de usuarios que solo han hecho cuatro resena

cantidad_resenas    1213
dtype: int64

Me he dado cuenta de que los datos estan ordenados por numero de usuario y el promedio de peliculas que los usuarion han rankeado a sido de 95, pero hay un usuario que ha rankeado 18,276 peliculas. de momento no se como interpretar esta info, tambien hay 2129 usuarios que solo han rankeado esta pelicula, solo me queda pensar de que elimine el 13% de los datos  con el unico fundamento de tener un archivo final menos pesado.

In [23]:
all_csv.drop(list(range(1500000,2999999)), inplace=True) # Elimino el segundo csv
all_csv.drop_duplicates(inplace=True)

In [24]:
len(all_csv)

9524176

## Transformo el campo "timestamp" a un formato descente

In [25]:
all_csv.timestamp = all_csv["timestamp"].apply(lambda x : date.fromtimestamp(x))

In [26]:
all_csv.head()

userId  score   timestamp movieId
0       1    1.0  2015-03-09   as680
1       1    4.5  2015-03-09  ns2186
2       1    5.0  2015-03-09  hs2381
3       1    5.0  2015-03-09  ns3663
4       1    5.0  2015-03-09  as9500

## Exportamos los datos a un csv

In [27]:
all_csv.to_csv("Data_util/Ratings.csv", index=False)